benchmarks/nnls/solvers/cd.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+from numpy.linalg import norm
+
+from benchopt.base import BaseSolver
+from benchopt.util import safe_import
+
+
+with safe_import() as solver_import:
+    from scipy import sparse
+    from numba import njit
+
+
+if solver_import.failed_import:
+
+    def njit(f):  # noqa: F811
+        return f
+
+
+@njit
+def positive_proj(x):
+    if x > 0:
+        return x
+    else:
+        return 0
+
+
+class Solver(BaseSolver):
+    name = "cd"
+
+    install_cmd = 'pip'
+    requirements = ['numba']
+
+    def set_objective(self, X, y):
+        self.X, self.y = X, y
+
+        # Make sure we cache the numba compilation.
+        self.run(1)
+
+    def run(self, n_iter):
+        if sparse.issparse(self.X):
+            L = sparse.linalg.norm(self.X, axis=0) ** 2
+            self.w = self.sparse_cd(
+                self.X.data, self.X.indices, self.X.indptr, self.y,
+                L, n_iter)
+        else:
+            L = norm(self.X, axis=0) ** 2
+            self.w = self.cd(self.X, self.y, L, n_iter)
+
+    @staticmethod
+    @njit
+    def cd(X, y, L, n_iter):
+        n_features = X.shape[1]
+        R = np.copy(y)
+        w = np.zeros(n_features)
+        for _ in range(n_iter):
+            for j in range(n_features):
+                old = w[j]
+                w[j] = positive_proj(w[j] + X[:, j] @ R / L[j])
```

Comment:
Maybe you can just use max(0, ...) without defining a function

Reply:
+1

Reply:
done